phoenix-core/src/it/java/org/apache/phoenix/end2end/EmptyColumnIT.java

```diff
@@ -0,0 +1,925 @@
+	private void assertAllHBaseRowsHaveEmptyColumnCell(byte[] hbaseTableName, byte[] emptyColumnFamilyName, byte[] emptyColumnName, long minTimestamp, int expectedRows) throws IOException, SQLException {
+
+		try (Table tbl = driver.getConnectionQueryServices(getUrl(), TestUtil.TEST_PROPERTIES)
+							 .getTable(hbaseTableName)) {
+
+			Scan allRows = new Scan();
+			allRows.setMaxVersions(1);
+			allRows.setTimeRange(minTimestamp, minTimestamp + Integer.MAX_VALUE);
+			ResultScanner scanner = tbl.getScanner(allRows);
+			int numMatchingRows = 0;
+			for (Result result = scanner.next();  result != null; result = scanner.next()) {
+				numMatchingRows += result.containsColumn(emptyColumnFamilyName, emptyColumnName) ? 1 : 0;
+			}
+
+			assertEquals(String.format(
+					"Expected rows do match for table = %s at timestamp %d",
+					Bytes.toString(hbaseTableName),
+					minTimestamp), expectedRows, numMatchingRows);
+		}
+
+	}
+
+	/**
+	 * -----------------
+	 * Test methods
+	 * -----------------
+	 */
+
+
+
+	@Ignore("fails with java.lang.ArrayIndexOutOfBoundsException: -1 , fails for both Table and GlobalView indexes")
+	// https://issues.apache.org/jira/browse/PHOENIX-5317
+	// https://issues.apache.org/jira/browse/PHOENIX-5322
+	public void testWithBasicTenantViewAndTableIndex() throws Exception {
+
+		// Define the test schema.
+		OtherOptions otherOptions = new OtherOptions();
+		otherOptions.setTableCFs(Lists.newArrayList(null, "A", "B"));
+
+		TableOptions tableOptions = TableOptions.withDefaults();
+		tableOptions.getTablePKColumns().add("ID");
+		tableOptions.getTablePKColumnTypes().add("CHAR(15)");
+
+		GlobalViewOptions globalViewOptions = new GlobalViewOptions();
+		globalViewOptions.setGlobalViewColumns(GLOBAL_VIEW_COLUMNS);
+		globalViewOptions.setGlobalViewColumnTypes(COLUMN_TYPES);
+		otherOptions.setGlobalViewCFs(Lists.newArrayList(null, "A", "B"));
+
+		final SchemaBuilder schemaBuilder = new SchemaBuilder(getUrl());
+		schemaBuilder
+				.withTableOptions(tableOptions)
+				.withGlobalViewOptions(globalViewOptions)
+				.withSimpleTenantView()
+				.withOtherOptions(otherOptions)
+				.build();
+
+		schemaBuilder
+				.withTableIndexDefaults()
+				.build();
+
+		// Define the test data.
+		DataSupplier dataSupplier = new DataSupplier() {
+			@Override
+			public List<Object> getValues(int rowIndex) {
+				Random rnd = new Random();
+				String id = String.format("00A0y000%07d", rowIndex);
+				String col1 = String.format("a%05d", rowIndex + rnd.nextInt(MAX_ROWS));
+				String col2 = String.format("b%05d", rowIndex + rnd.nextInt(MAX_ROWS));
+				String col3 = String.format("c%05d", rowIndex + rnd.nextInt(MAX_ROWS));
+				String col4 = String.format("d%05d", rowIndex + rnd.nextInt(MAX_ROWS));
+				String col5 = String.format("e%05d", rowIndex + rnd.nextInt(MAX_ROWS));
+				String col6 = String.format("f%05d", rowIndex + rnd.nextInt(MAX_ROWS));
+				return Lists.newArrayList(new Object[] {id, col1, col2, col3, col4, col5, col6});
```

Comment:
what's the format of `00A0y000%07d`? 
why we are using `a%05d, b%05d,...f%05d` instead of random directly? phoenix format? where can I get that information, thanks

Reply:
Nothing specific about the format "00A0y000%07d", Just follows the SFDC 15 char id format.
Again nothing specific about "a%05d", just a prefix before the random number so that the data is easily identifiable when debugging or query thru hbase shell.